### PR TITLE
fix: Empty pkg.repository renders a blank page

### LINKED ISSUE #780: when repository property of package.json is empty, the package detail page would render white blank
**Describe the bug**
repository property of package.json is empty, the package detail page would render white blank
**To Reproduce**
1. set package.json
{
 ...
  repository: {}  // <----no git url
...
}
2. publish
3. go to detail page 
/#/detail/[package-page]
4 the page is empty 

**Expected behavior**
remove  repository is ok 
**Screenshots**

### PR DESCRIPTION
**Pull request by lgaitan**
Type: **Bug**

Description:
If the published package contains a empty `repository` field, the Maintainers component will try to access a property on an `undefined` variable, resulting in an error.

This PR also fixes another error on the PackageSidebar.Maintainers component that also throws an error when the `author.name` property is missing

Resolves #780

### CODE REVIEW

**Comment by juanpicado** [comment]
We should update unit test or add a new scenario there.

**Comment by juanpicado** [comment]
I'd suggest also update `yarn test:e2e` test when we modify any UI feature.  https://github.com/verdaccio/verdaccio/wiki/Running-and-Debugging-tests#e2e-test

File: src/webui/src/components/PackageSidebar/modules/Maintainers/index.jsx [hunk 1]
```diff
@@ -69,6 +69,11 @@ export default class Maintainers extends React.Component {
   render() {
     let author = this.author;
 
+    // There is no author info on `package.json`
+    if (!author.name) {
```
Comment (juanpicado) [inline src/webui/src/components/PackageSidebar/modules/Maintainers/index.jsx]: Flow will detect this as possible undefined,

Comment (juanpicado) [reply src/webui/src/components/PackageSidebar/modules/Maintainers/index.jsx]: `this.author` may be clearly undefined

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx [hunk 2]
```diff
@@ -23,17 +23,17 @@ export default class Infos extends React.Component {
...
-    infoObj.url = this.normalizeGitUrl(infoObj);
+    infoObj.url = this.normalizeGitUrl(infoObj.url);
 
     return infoObj;
   }
 
-  normalizeGitUrl(infoObj) {
-    return infoObj.url.replace(/^git\+/, '');
+  normalizeGitUrl(gitUrl) {
+    return gitUrl ? gitUrl.replace(/^git\+/, '') : '';
```
[the remark this answers is not in the document]
Comment (ayusharma) [reply src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: You can add a unit test for this.

File: src/webui/src/components/PackageSidebar/modules/Maintainers/index.jsx (hunk 1 above)
Comment (juanpicado) [reply src/webui/src/components/PackageSidebar/modules/Maintainers/index.jsx]: ping @lgaitan

**Comment by lgaitan** [comment]
I'm sorry, i was away for personal issues.
I'm gonna fix the requested changes and submit again

**Comment by lgaitan** [comment]
do i still need to make any further changes?

i was looking into the `master` branch, and the changes i've proprosed in this PR were overwritten by another commit.

**Comment by juanpicado** [comment]
@ayusharma does #833 solves https://github.com/verdaccio/verdaccio/issues/780 ?

**Comment by ayusharma** [comment]
Yes, #833 solves the problem.

**Comment by ayusharma** [comment]
@juanpicado I am closing this issue. @lgaitan Thank you so much for your time.

Fixed in #833

**Pull request closed without merging**